modules/sagemaker/sagemaker-custom-kernel/modulestack.yaml

```diff
@@ -0,0 +1,61 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: This stack deploys a Module specific IAM permissions
+
+Parameters:
+  ProjectName:
+    Type: String
+    Description: The name of the project
+  DeploymentName:
+    Type: String
+    Description: The name of the deployment
+  ModuleName:
+    Type: String
+    Description: The name of the Module
+  RoleName:
+    Type: String
+    Description: The name of the IAM Role
+  StudioExecutionRoleArn:
+    Type: String
+    Description: The name of the Sagemaker Studio IAM Role
+  ECRRepoName:
+    Type: String
+    Description: The name of the ECR repository
+  StudioDomainId:
+    Type: String
+    Description: The id of the SageMaker Studio Domain
+
+Resources:
+  Policy:
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Effect: Allow
+            Action:
+              - "sagemaker:AddTags"
+              - "sagemaker:DescribeDomain"
+              - "sagemaker:UpdateDomain"
+            Resource:
+              - !Sub "arn:aws:sagemaker:${AWS::Region}:${AWS::AccountId}:domain/${StudioDomainId}"
+          - Effect: Allow
+            Action:
+              - "iam:PassRole"
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/${ProjectName}-${DeploymentName}-${ModuleName}-image-role"
+              - !Ref StudioExecutionRoleArn
+          - Action:
+              - "ecr:*LayerUpload"
```

Comment:
Curious to know why do you need ECR IAM permissions? I have noticed you are building images using `CDK Lib`

Reply:
Thanks!